revise tutorial

Clownshift opened the pull request:
Supposed to fix #3 : 
I changed the line with `hydrogenize.sh` to expicitly use bash, rather than execute the file.

Easy fix for #4 :
Explicitly only use a single mpi-thread. with `-ntmpi 1`. I hope this works on all machines.

About #5 :
I tried to improve reproducibility of the simulations, to enable the detailed discussion of the trajectories, but in the end it is still not exactly reproducible. Therefore, I rewrote parts of the discussion, since the precise mentioning of the certain frames in the simulations is not meaningful...

Besides that I added a vmd scripts to visualize the simulation. Many users might be interested in that, I think.

Commit:
vmd-script, random seeds, rewriting

I added a vmd scripts to visualize the simulation. Many users might be
interested in that, I think.
I tried to improve reproducibility of the simulations, to enable the
detailed discussion of the trajectories, but in the end it is still not
exactly reproducible. Therefore, I rewrote parts of the discussion,
since the precise mentioning of the certain frames in the simulations
is not meaningful...

Clownshift commented:
@daandtu Note that I am only pushing this to the Zacharias-lab not to your original repo. We may discuss, if you prefer it otherwise...?

daandtu commented on tutorial/simulation.mdp:
The `simulation.mdp` file in the main `tutorial` directory should be the simple/short version without all default values. The automatically generated `simulation.mdp` file, including all default values, should only be present in the `output` or, here, the `output-worked` directory.

Clownshift replied:
Yes, you are right. Actually I tried for quite a while to get reproducible results for the simulations. To this end, the user may specify random seeds and use the `--reprod` in mdrun. Anyways, gromacs does reproduce the results, so I guess we may revert to a more reduced `mdp` file. I'll take care of that...

daandtu commented on tutorial/simulation.vmd:
I assume this file was generated automatically by VMD, and we don't really need it?

Clownshift replied:
No, I provided this file. I think most users are not used to visualizing coarse-grained simulation. Therefore, I thought this might be a convenient file to visualize the trajectory from the tutorial.

daandtu commented on tutorial/tutorial.ipynb:
```diff
@@ -16,15 +16,19 @@
     "- Trajectory processing\n",
-    "- Simulation evaluation"
+    "- Simulation Analysis\n",
+    "- (Visualization of trajectory)*\n",
```
We could add _"Visualization of the trajectory with VMD"_ without the *-annotation and add _VMD_ to the list of prerequisites. I think the average user of the tutorial can use another program if _VMD_ is unavailable.

Clownshift replied:
Good, idea. I'll quickly do that.

Clownshift replied:
should be resolved here:
https://github.com/ZachariasLab/GoCa/pull/6/commits/d8230322ca4a993535705239d7f0f3316b764f99

daandtu commented on tutorial/tutorial.ipynb:
```diff
@@ -43,25 +47,15 @@
+   "id": "d518022c",
    "metadata": {},
    "source": [
-    "The next cell contains the path to the root directory of the GoCa repository. If you move this notebook to a new location, you should add the path to the repository in the following cell. Make sure you also move the included `.mdp` files."
```
Switching the setup of the GoCa repository path and the installation does not work since the `fp` variable is used during the installation step.

Clownshift replied:
What do you think about removing the installation for the tutorial and instead only put a note there: 
`Install GoCa, as described in the Readme.md`

daandtu reviewed (changes requested):
Good suggestions!

Is it on purpose that the notebook does not contain any outputs? I thought having the expected output, e.g. the plots, provided with the notebook, would be nice.

Regarding #3, #4, and #5: All good ideas!!!

After discussing my suggestions, I can also make the changes to the `revise tutorial`-branch myself such that you can do another review.

Clownshift commented:
> Is it on purpose that the notebook does not contain any outputs? I thought having the expected output, e.g. the plots, provided with the notebook, would be nice.

Yes, I did that on purpose, because like that the version control of the notebook is quite straight-forward: After clearing all output, it should be in the same state as in the repo.
You have a point though, that the users would have an idea of the expected output, if we kept the plots in there.
However, since we already provide expected output trajectories, we could also provide the figures there. Or lead the data from there and create plots from that? What do you think?


> After discussing my suggestions, I can also make the changes to the `revise tutorial`-branch myself such that you can do another review.

Maybe, we can do that. I can take care of a few minor steps quickly now.

Commit:
minor change: change TOC headline

rewrote the bullet point for the visualization of vmd

Commit:
simplify simulation.mdp

reverted changes in this file to make it more simple again.
In comparison to main branch, the only difference are the fixed random
seeds.

Commit:
fewer steps in tutorual sim.

Commit:
remove installation from tutorial

We removed the explicit installation from the tutorial and refer to the
README for installation instructions instead.

Commit:
add example figures

I generated figures from the trajectory in the output-worked path.
added those there. In upcoming commits, These should be included in the tutorial notebook for
reference.

daandtu committed:
Add example images to tutorial notebook

Commit:
comment on special case: chain permutation

It actually happened to me now once that the structure assembled in a
different permutation as in the crystal structure. Therefore, the RMSD
of the complex never approached zero. I added a few sentences in the
tutorial notebook to explain this eventual case.

Commit:
remove cell-ouputs...

Commit:
remove typo

daandtu commented:
The RMSD value for permutated chains is a good point. I forgot about that.
I actually implemented a permutation invariant RMSD calculation ([here](https://github.com/ZachariasLab/GoCa/blob/bf587ef57bbd8c3f74e385794720130f599ae482/evaluation/analysis.py#L520C9-L520C40), so we could change the tutorial to include that function instead of the normal RMSD calculation.
I can do the change later today!

Clownshift commented:
> The RMSD value for permutated chains is a good point. I forgot about that. I actually implemented a permutation invariant RMSD calculation ([here](https://github.com/ZachariasLab/GoCa/blob/bf587ef57bbd8c3f74e385794720130f599ae482/evaluation/analysis.py#L520C9-L520C40), so we could change the tutorial to include that function instead of the normal RMSD calculation. I can do the change later today!

The permutated assembly only occurs quite rarely, actually. (At least with the current fixed seed and so on.) I executed the notebook probably five or six times, but it only happened once. 

Honestly, I think it is a super cool result, because it really clarifies the chain permutation functionality. I am considering to include a second output-worked directory for the alternative result. (I didn't delete them, yet) What do you think?

daandtu commented:
> The permutated assembly only occurs quite rarely, actually

That's interesting. Since we randomly place the subunits in the simulation box each permutation (of the two possible) should occur equally likely (maybe not with the fixed seeds).

> Honestly, I think it is a super cool result, because it really clarifies the chain permutation functionality. I am considering to include a second output-worked directory for the alternative result. (I didn't delete them, yet) What do you think?

We could still include one simulation result with the "original" permutation and one with the "new" permutation, but I think we don't need a complete second output-worked directory, we can just have the files in the same directory with similar names and mention that in the notebook.
Nevertheless, it is probably good to include the permutation invariant RMSD calculation method in the notebook.

Clownshift commented:
> That's interesting. Since we randomly place the subunits in the simulation box each permutation (of the two possible) should occur equally likely (maybe not with the fixed seeds).

I fixed the seeds for the `gmx insert-molecules` command and that actually always leads to the same starting geometry now. (This is reproducible)

Commit:
add figs for permutated assembly

also added the trajectory with the alternative assembly in the
output-worked path.

Clownshift commented:
> Nevertheless, it is probably good to include the permutation invariant RMSD calculation method in the notebook.

I added the  trajectory with the alternative transition pathway. You may use your invariant RMSD calculation on that. I also added pictures of the the two possible assemblies. Not 100% happy with  the html inclusion in the ipynb, but I think it's functional...

daandtu committed:
Add comment for permutation invariant RMSD calculation

daandtu commented:
The permutation invariant RMSD calculation is too slow, i.e., it is not practical to use it for the whole trajectory. Therefore, I left the normal RMSD calculation and added a comment, that this function exists.
